paymentsheet/src/main/java/com/stripe/android/paymentelement/embedded/SharedPaymentElementViewModel.kt

```diff
@@ -69,12 +75,19 @@ import kotlin.reflect.KClass
 
 @OptIn(ExperimentalEmbeddedPaymentElementApi::class)
 internal class SharedPaymentElementViewModel @Inject constructor(
+    confirmationHandlerFactory: ConfirmationHandler.Factory,
+    @IOContext ioContext: CoroutineContext,
     private val configurationHandler: EmbeddedConfigurationHandler,
     private val paymentOptionDisplayDataFactory: PaymentOptionDisplayDataFactory,
 ) : ViewModel() {
     private val _paymentOption: MutableStateFlow<PaymentOptionDisplayData?> = MutableStateFlow(null)
     val paymentOption: StateFlow<PaymentOptionDisplayData?> = _paymentOption.asStateFlow()
 
+    val confirmationHandler = confirmationHandlerFactory.create(viewModelScope + ioContext)
```

Comment:
This is ONLY used to store a reference to the `confirmationHandler`. Ideally we could do it outside of here, but the dependency injection made it hard, and this seemed like a fair compromise.